tests/functional/cnv/test_vms.py

```diff
@@ -0,0 +1,144 @@
+
+        # 2. Validate data integrity using md5sum.
+        file_name = "/tmp/dd_file"
+        vm_filepath = "/home/admin/dd_file1_copy"
+
+        # Create file locally
+        file_path = str(os.path.abspath(file_name))
+        cmd = f"dd if=/dev/zero of={file_path} bs=1M count=2048"
+        run_cmd(cmd)
+
+        # Calculate the MD5 checksum
+        if file_path:
+            md5sum_on_local = self.get_md5sum(file_path)
+            if md5sum_on_local:
+                print(f"MD5 checksum of the file: {md5sum_on_local}")
+
+        # Copy local file to all vms
+        for vm_obj in self.vm_objs_aggr:
+            vm_obj.scp_to_vm(
+                local_path=file_path,
+                vm_username=None,
+                identity_file=None,
+                vm_dest_path=vm_filepath,
+                recursive=False,
+            )
+        for vm_obj in self.vm_objs_def:
+            vm_obj.scp_to_vm(
+                local_path=file_path,
+                vm_username=None,
+                identity_file=None,
+                vm_dest_path=vm_filepath,
+                recursive=False,
+            )
+
+        # Take md5sum of copied file and compare with md5sum taken locally
+        for vm_obj in self.vm_objs_aggr:
+            md5sum_on_vm = cal_md5sum_vm(vm_obj, vm_filepath, username=None)
+            assert (
+                md5sum_on_vm == md5sum_on_local
+            ), f"md5sum has not changed after copying file on {vm_obj.name}"
+
+        for vm_obj in self.vm_objs_def:
+            md5sum_on_vm = cal_md5sum_vm(vm_obj, vm_filepath, username=None)
+            assert (
+                md5sum_on_vm == md5sum_on_local
+            ), f"md5sum has not changed after copying file on {vm_obj.name}"
+
+        # 3.Verify PV Keyrotation.
+        for vm in self.vm_objs_def:
+            pvk_obj = PVKeyrotation(self.sc_obj_def_compr)
+            assert pvk_obj.wait_till_keyrotation(
+                vm.pvc_obj.get_pv_volume_handle_name
+            ), f"Failed to rotate Key for the PVC {vm.pvc_obj.name}"
+
+        for vm in self.vm_objs_aggr:
+            pvk_obj = PVKeyrotation(self.sc_obj_aggressive)
+            volume_name = vm.pvc_obj.get().get("spec").get("volumeName")
+            for line in run_oc_command(
+                f"describe pv {volume_name}", namespace=vm.namespace
+            ):
+                if "VolumeHandle:" in line:
+                    volume_handle = line.split()[1]
+                    break
+            if volume_handle is None:
+                logger.error(f"Cannot get volume handle for pv {volume_name}")
+                raise Exception("Cannot get volume handle")
+            assert pvk_obj.wait_till_keyrotation(
+                volume_handle
+            ), f"Failed to rotate Key for the PVC {vm.pvc_obj.name}"
+
+        # 4.Stop all VMs
+        for vm_obj in self.vm_objs_def + self.vm_objs_aggr:
+            vm_obj.stop()
```

Comment:
I don't this part of the test case? do we need this, shouldn't the fixture teardown take care of stopping the VM's?